specification/marketplacecatalog/data-plane/Search/preview/2022-09-25-preview/search.json

```diff
@@ -0,0 +1,2229 @@
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ErrorResponse": {
+      "description": "The search error response object",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The search error response details",
+          "$ref": "#/definitions/ErrorResponseDetails"
+        }
+      }
+    },
+    "ErrorResponseDetails": {
+      "description": "The search error response details object",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code description. Such as code=InternalError",
+          "type": "string"
+        },
+        "message": {
+          "description": "The error message. Such as message=Internal server error",
+          "type": "string"
+        }
+      }
+    },
+    "PricingType": {
+      "type": "string",
+      "description": "Pricing type",
+      "enum": [
+        "Free",
+        "FreeTrial",
+        "Byol",
+        "Payg",
+        "Ri"
+      ],
+      "x-ms-enum": {
+        "name": "PricingType",
+        "modelAsString": true
+      }
+    },
+    "RatingBucket": {
+      "type": "string",
+      "description": "Azure portal rating bucket to search by. Such as 'Above1'",
+      "enum": [
+        "AboveOne",
+        "AboveTwo",
+        "AboveThree",
+        "AboveFour"
+      ],
+      "x-ms-enum": {
+        "name": "RatingBucket",
+        "modelAsString": true
+      }
+    },
+    "VmSecurityType": {
+      "type": "string",
+      "description": "Vm security type, Possible values: Trusted, Confidential, None.",
+      "enum": [
+        "None",
+        "Trusted",
+        "Confidential"
+      ],
+      "x-ms-enum": {
+        "name": "VmSecurityType",
+        "modelAsString": true
+      }
+    },
+    "SuggestionType": {
+      "type": "string",
+      "description": "The type of the suggestion",
+      "enum": [
+        "WordSearch",
+        "Entity"
+      ],
+      "x-ms-enum": {
+        "name": "SuggestionType",
+        "modelAsString": true
+      }
+    },
+    "FacetsItem": {
+      "type": "object",
+      "description": "The facets item",
+      "properties": {
+        "facetValues": {
+          "type": "object",
+          "description": "The facet values",
+          "additionalProperties": {
+            "$ref": "#/definitions/FacetValue"
+          }
+        }
+      }
+    },
+    "FacetValue": {
+      "type": "object",
+      "description": "The facet value",
+      "properties": {
+        "count": {
+          "description": "The facet count",
+          "type": "integer",
+          "format": "int64"
+        }
+      }
+    },
+    "FacetsResponse": {
+      "type": "object",
+      "description": "The facets response",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "description": "The facet items results",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
We should explicitly state which facet names are valid here rather than using `additionalProperties`. As expressed now, it is under-specifying the actual API.

Reply:
The valid facet names come from the facet request enum:

         {
           "type": "array",
           "name": "facets",
           "in": "query",
           "description": "Array facets to facet by, if none provided then no facets will return. Such as facets=PricingTypes,operatingSystems. \n- `All`: All fields.\n- `Popularity`: Popularity.\n- `ApplicableProducts`: Applicable Products.\n- `CategoryIds`: Category Ids.\n- `Market`: Market.\n- `LinkedAddIns`: Linked AddIns.\n- `SupportedProducts`: Supported Products.\n- `HideKeys`: Hide Keys.\n- `PublisherId`: Publisher Id.\n- `CspStates`: Csp States.\n- `DisplayName`: Display Name.\n- `AzureBenefit`: Azure Benefit.\n- `Badges`: Badges.\n- `SmallIconUri`: Small Icon Uri.\n- `MediumIconUri`: Medium Icon Uri.\n- `LargeIconUri`: Large Icon Uri.\n- `WideIconUri`: Wide Icon Uri.\n- `IndustryCloud`: Industry Cloud.\n- `PublisherType`: Publisher Type.\n- `PublishingState`: Publishing State.\n- `Language`: Language.\n- `UniqueProductId`: Unique Product Id.\n- `ProductType`: Product Type.\n- `Plans`: Plans.\n- `OperatingSystems`: Operating Systems.\n- `PricingTypes`: Pricing Types.\n- `PublisherDisplayName`: Publisher Display Name.\n- `Summary`: Summary.\n- `VmImageGenerations`: Vm Image Generations.\n- `VmSecurityTypes`: Vm Security Types.\n- `VmArchitectureTypes`: Vm Architecture Types.\n- `Description`: Description.\n- `RatingBuckets`: Rating Buckets.\n- `RatingAverage`: Rating Average.",



           "collectionFormat": "csv",



           "items": {



             "type": "string",



             "enum": [



               "All",



               "Popularity",



               "ApplicableProducts",



               "CategoryIds",



               "Market",



               "LinkedAddIns",



               "SupportedProducts",



               "HideKeys",



               "PublisherId",



               "CspStates",



               "DisplayName",



               "AzureBenefit",



               "Badges",



               "SmallIconUri",



               "MediumIconUri",



               "LargeIconUri",



               "WideIconUri",



               "IndustryCloud",



               "PublisherType",



               "PublishingState",



               "Language",



               "UniqueProductId",



               "ProductType",



               "Plans",



               "OperatingSystems",



               "PricingTypes",



               "PublisherDisplayName",



               "Summary",



               "VmImageGenerations",



               "VmSecurityTypes",



               "VmArchitectureTypes",



               "Description",



               "RatingBuckets",



               "RatingAverage"
             ],
             "x-ms-enum": {
               "name": "SearchV2FieldName",
               "modelAsString": true
             }
           }

The response just reflects the facets that were asked in the request